documentation/SolutionsRegistryDataSource.md

```diff
@@ -0,0 +1,122 @@
+# Solutions Registry Data Source
+
+The solutions registry data source provides a RESTful means of fetching
+solutions registries via the [FlowManager](FlowManager.md) in order to determine
+which solutions are available and appropriate for a user's preferences.  A
+solution is an application, such as the NVDA screen reader, or an operating
+system feature, such as the Windows high contrast theme set using a system
+control panel.  Each solution supports a set of preferences and describes how
+to configure the solution, launch it, reset it, and stop it.
+
+The solutions are listed in `JSON` files called "solutions registries".  The
+structure of a solutions registry is documented in
+[SolutionsRegistryFormat](SolutionsRegistryFormat.md).  Since solutions are
+frequently specific to an operating system, or "platform", there are separate
+solution registries for each platform.  Examples of platforms are Windows,
+GNOME-Linux, MacOS, and Android.
+
+The SolutionsRegistryDataSource is a component that loads and caches
+solutions registries when the GPII starts up, and then serves solutions to the
+rest of the system upon request.  The flow manager coordinates retrieval of
+solutions, user preferences, device information, and so on, passing these to the
+[MatchMaker Framework](MatchMakerFramework.md) and [LifecycleManager](LifecycleManager.md).
+
+There are two initialization workflows with respect to solutions registries
+depending on whether the flow manager is running in the cloud or locally on the
+client device.  This are described in the following two sections.
+
+## Cloud Based Flow Manager
```

Comment:
Nice writing. Thanks, @klown.

The section title of "Cloud Based Flow Manager" might be misleading. What about "Cloud Based Solution Registry Data Source"? The section content then describes it is running as a sub-component in the Cloud Based Flow Manager.

The same comment applies to the next section title "Local Flow Manager".

Reply:
The paragraph that precedes the section is meant to introduce the next two sections as two different scenarios and describe the way the SRDS is initialized in the context of the CBFM and the LFM. I've tried to make that more explicit, and changed the headings.  Hopefully it's clearer.